test/jdk/sun/java2d/cmm/ColorConvertOp/ColCvtAlphaDifferentSrcDst.java

```diff
@@ -0,0 +1,170 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AlphaComposite;
+import java.awt.Graphics2D;
+import java.awt.color.ColorSpace;
+import java.awt.image.BufferedImage;
+import java.awt.image.ColorConvertOp;
+
+import static java.awt.image.BufferedImage.TYPE_3BYTE_BGR;
+import static java.awt.image.BufferedImage.TYPE_4BYTE_ABGR;
+import static java.awt.image.BufferedImage.TYPE_4BYTE_ABGR_PRE;
+import static java.awt.image.BufferedImage.TYPE_BYTE_BINARY;
+import static java.awt.image.BufferedImage.TYPE_BYTE_GRAY;
+import static java.awt.image.BufferedImage.TYPE_BYTE_INDEXED;
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB_PRE;
+import static java.awt.image.BufferedImage.TYPE_INT_BGR;
+import static java.awt.image.BufferedImage.TYPE_INT_RGB;
+import static java.awt.image.BufferedImage.TYPE_USHORT_555_RGB;
+import static java.awt.image.BufferedImage.TYPE_USHORT_565_RGB;
+import static java.awt.image.BufferedImage.TYPE_USHORT_GRAY;
+
+/*
+ * @test
+ * @bug 8012229
+ * @summary one more test to check the alpha channel
+ */
+public final class ColCvtAlphaDifferentSrcDst {
+
+    private static int WIDTH = 256;
+    private static int HEIGHT = 256;
+
+    public static void main(String[] args) throws Exception {
+        differentToOpaqueDst();
+        differentToTransparentDst(TYPE_INT_ARGB);
+        differentToTransparentDst(TYPE_4BYTE_ABGR);
+        differentToTransparentDst(TYPE_INT_ARGB_PRE);
+    }
+
+    /**
+     * Various types of source images transform to the opaque destination, the
+     * result should be the same.
+     */
+    private static void differentToOpaqueDst() {
+        opaqueDst(TYPE_INT_ARGB, TYPE_INT_RGB);
+        opaqueDst(TYPE_INT_ARGB, TYPE_INT_BGR);
+        opaqueDst(TYPE_4BYTE_ABGR, TYPE_INT_BGR);
+
+        // It is unclear how to hangle pre colors in the opaque DST
+        //opaqueDst(TYPE_INT_ARGB_PRE, TYPE_4BYTE_ABGR_PRE);
+        //opaqueDst(TYPE_4BYTE_ABGR_PRE, TYPE_INT_BGR);
+    }
```

Comment:
Actually, according to the spec, we should transform pre-alpha to alpha -> then apply ICC transform -> then transform pre-alpha-> alpha.
See class description of the:
https://docs.oracle.com/javase/7/docs/api/java/awt/image/ColorConvertOp.html

But it looks like implementation does something different.